oak-segment-remote/src/main/java/org/apache/jackrabbit/oak/segment/remote/AbstractRemoteSegmentArchiveWriter.java

```diff
@@ -150,6 +150,12 @@ public void flush() throws IOException {
         }
     }
 
+    @Override
+    public void flush(Runnable callback) throws IOException {
```

Comment:
I would use here the same parameter name `onSuccess` than in the interface, instead of `callback`.